android/src/com/frostwire/android/core/FileDescriptor.java

```diff
@@ -82,6 +84,39 @@ public FileDescriptor(int id, String artist, String title, String album, String
         ensureCorrectMimeType(this);
     }
 
+    public FileDescriptor(Bundle bundle) {
+        this(bundle.getInt("id"),
```

Comment:
I guess this is what you meant, but I'm not sure, as I end up repeating code in the second constructor

https://github.com/frostwire/frostwire/pull/437/commits/551068ab6d5be6f2bf9f52ff64f5bd63b8c6747f#diff-4d8bf0d1f6b9710a2c1c6bb359420232R88